pkg/doc/bbs/bbs12381g2pub/bbs.go

```diff
@@ -116,55 +126,58 @@ func (bbs *BBSG2Pub) SignWithKey(messages [][]byte, privKey *PrivateKey) ([]byte
 		}
 	}
 
-	e, err := bls.RandFR(rand.Reader)
+	e, err := createRandSignatureFr()
 	if err != nil {
 		return nil, fmt.Errorf("create signature.E: %w", err)
 	}
 
-	s, err := bls.RandFR(rand.Reader)
+	s, err := createRandSignatureFr()
 	if err != nil {
 		return nil, fmt.Errorf("create signature.S: %w", err)
 	}
 
-	b, err := computeB(s, messagesFr, pubKey)
+	b, err := bbs.computeB(s, messagesFr, pubKey)
 	if err != nil {
 		return nil, fmt.Errorf("compute B point: %w", err)
 	}
 
-	exp := privKey.GetFRElement().Copy()
-	exp.AddAssign(e)
-	sig := b.MulFR(exp.Inverse().ToRepr())
+	exp := bls12381.NewFr().Set(privKey.FR)
+	exp.Add(exp, e)
+	exp = exp.Inverse()
```

Comment:
@kdimak @llorllale @Baha-sk Looks like the upstream function signature (that was just added) is a bit different:
https://github.com/kilic/bls12-381/blob/a288617c07f1bd60613c43dbde211b4a911e4791/fr.go#L257

Reply:
I will update our fork and use the original kilic function (and double-check in works as expected of course).